modules/images/webp-uploads/load.php

```diff
@@ -42,60 +46,55 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 		return $metadata;
 	}
 
-	$dirname     = pathinfo( $file, PATHINFO_DIRNAME );
-	$image_sizes = array();
-	if ( array_key_exists( 'sizes', $metadata ) && is_array( $metadata['sizes'] ) ) {
-		$image_sizes = $metadata['sizes'];
-	}
-
-	foreach ( wp_get_registered_image_subsizes() as $size_name => $properties ) {
-		// This image size does not exist on the defined sizes.
-		if ( ! isset( $image_sizes[ $size_name ] ) || ! is_array( $image_sizes[ $size_name ] ) ) {
+	$dirname = pathinfo( $file, PATHINFO_DIRNAME );
+	foreach ( $metadata['sizes'] as $size_name => $properties ) {
+		// This image size is not defined or not an array.
+		if ( ! isset( $metadata['sizes'][ $size_name ] ) || ! is_array( $metadata['sizes'][ $size_name ] ) ) {
```

Comment:
At least the first check here is now redundant since we're iterating through exactly this array.
```suggestion
		if ( ! is_array( $properties ) ) {
```

Reply:
Good point.